### PR TITLE
Added step-1 validation for description fields and document changes

### PR DESCRIPTION
**Pull request by lavanya-bmw**

### CODE REVIEW

**Review by oyo — CHANGES_REQUESTED** [review]

File: cx-portal/src/features/appManagement/slice.ts [hunk 1]
```diff
@@ -37,6 +37,9 @@ const managementSlice = createSlice({
     decrement: (state) => {
       state.currentActiveStep -= 1
     },
+    setCurrentActiveStep: (state) => {
+      state.currentActiveStep = 1
```
[the remark this answers is not in the document]
Comment (lavanya-bmw) [reply cx-portal/src/features/appManagement/slice.ts:41]: @oyo After completing all steps and submitting the app release process successfully and if user again goes to app release process, the user should be able to see step1 again. (This is not related to the ticket 1624, but a small adjustment added to this PR)

File: cx-portal/src/components/pages/AppReleaseProcess/components/AppMarketCard/index.tsx [hunk 2]
```diff
@@ -476,8 +476,8 @@ export default function AppMarketCard() {
                               'content.apprelease.appReleaseForm.validCharactersIncludes'
                             )} ${
                               item === 'shortDescriptionEN'
-                                ? 'A-Za-z0-9.: @&,'
-                                : 'A-Za-z0-9.: @&,äüö'
+                                ? 'A-Za-z0-9.: @&_-,'
+                                : 'A-Za-z0-9.: @&_-,äüö'
```
[the remark this answers is not in the document]
Comment (lavanya-bmw) [reply cx-portal/src/components/pages/AppReleaseProcess/components/AppMarketCard/index.tsx:480]: @jjeroch Can you please check and confirm me if these characters also to be included in the validations?

[the remark this answers is not in the document]
Comment (jjeroch) [reply cx-portal/src/components/pages/AppReleaseProcess/components/AppMarketCard/index.tsx:480]: Is this the short description? If so, I would not open up the characters that much. Its only one sentence, so we should keep it limited to some key characters. I did do a mix from the actual validation and the proposal from martin - ideally we do following for the short description:
`a-zA-ZÀ-ÿ0-9 !?@&#'"()_-=/*.,;:`

and lets use the full list `a-zA-ZÀ-ÿ0-9 !?@&#'"()[]_-+=<>/*.,;:` for the long description field (y)

File: cx-portal/src/types/Patterns.ts [hunk 3]
```diff
@@ -33,8 +33,8 @@ export const Patterns = {
   appMarketCard: {
     appTitle: /^([A-Za-z.:_@&0-9 -]){5,40}$/,
     appProvider: /^([A-Za-z ]){3,30}$/,
-    shortDescriptionEN: /^([A-Za-z.:@0-9& ,]){10,255}$/,
-    shortDescriptionDE: /^([A-Za-z.:@0-9& ,äüö]){10,255}$/,
+    shortDescriptionEN: /^([A-Za-z.:@0-9&_ ,-]){10,255}$/,
+    shortDescriptionDE: /^([A-Za-z.:@0-9&_ ,äüö-]){10,255}$/,
```
[the remark this answers is not in the document]
Comment (jjeroch) [reply cx-portal/src/types/Patterns.ts:37]: see comment above :) thanks for the suggestion

**Review by manojava-gk — DISMISSED** [review]
lftm

**Review by oyo — APPROVED** [review]
looks good now

**Pull request merged**